core/types/types.cc

```diff
@@ -172,8 +172,11 @@ TypePtr Types::dropSubtypesOf(Context ctx, const TypePtr &from, SymbolRef klass)
             }
         },
         [&](ClassType *c) {
+            auto cdata = c->symbol.data(ctx);
             if (c->isUntyped()) {
                 result = from;
+            } else if (cdata->isClassSealed() && (cdata->isClassAbstract() || cdata->isClassModule())) {
```

Comment:
this case needs `cdata`, while the next one doesn't. This suggests that the next case should be arlier as it can be checked faster